Notify/readers.ts

```diff
@@ -0,0 +1,159 @@
+  RetrievedService
+>;
+export const getService = (serviceModel: ServiceModel): ServiceReader => (
+  serviceId
+): ReturnType<ServiceReader> =>
+  pipe(
+    serviceModel.findOneByServiceId(serviceId),
+    TE.mapLeft(_ =>
+      ResponseErrorInternal("Error while retrieving the service")
+    ),
+    TE.chainW(
+      TE.fromOption(() =>
+        ResponseErrorNotFound(
+          "Service not found",
+          `Service ${serviceId} was not found in the system.`
+        )
+      )
+    )
+  );
+
+const getMessageMetadata = (messageModel: MessageModel) => (
+  fiscalCode: FiscalCode,
+  messageId: NonEmptyString
+): TE.TaskEither<
+  IResponseErrorInternal | IResponseErrorNotFound,
+  RetrievedMessage
+> =>
+  pipe(
+    messageModel.findMessageForRecipient(fiscalCode, messageId),
+    TE.mapLeft(_ =>
+      ResponseErrorInternal("Error while retrieving the message metadata")
+    ),
+    TE.chainW(
+      TE.fromOption(() =>
+        ResponseErrorNotFound(
+          "Message not found",
+          `Message ${messageId} was not found for the given Fiscal Code`
+        )
+      )
+    )
+  );
+
+const getMessageContent = (
+  messageModel: MessageModel,
+  blobService: BlobService
+) => (
+  messageId: NonEmptyString
+): TE.TaskEither<
+  IResponseErrorInternal | IResponseErrorNotFound,
+  MessageContent
+> =>
+  pipe(
+    messageModel.getContentFromBlob(blobService, messageId),
+    TE.mapLeft(_ =>
+      ResponseErrorInternal("Error while retrieving the message")
+    ),
+    TE.chainW(
+      TE.fromOption(() =>
+        ResponseErrorNotFound(
+          "Message content not found",
+          `Content of message ${messageId} was not found for the given Fiscal Code`
+        )
+      )
+    )
+  );
+
+export type MessageWithContentReader = (
+  fiscalCode: FiscalCode,
+  messageId: NonEmptyString
+) => TE.TaskEither<
+  IResponseErrorInternal | IResponseErrorNotFound,
+  RetrievedMessageWithContent
+>;
+/**
+ * Retrieve a message with content
+ * Return an IResponseError_ otherwise
+ */
+export const getMessageWithContent = (
+  messageModel: MessageModel,
+  blobService: BlobService
+): MessageWithContentReader => (
+  fiscalCode,
+  messageId
+): ReturnType<MessageWithContentReader> =>
+  pipe(
+    {
+      content: getMessageContent(messageModel, blobService)(messageId),
+      metadata: getMessageMetadata(messageModel)(fiscalCode, messageId)
+    },
+    AP.sequenceS(TE.ApplicativePar),
+    TE.map(({ metadata, content }) => ({
+      ...metadata,
+      content,
+      kind: "IRetrievedMessageWithContent"
+    }))
+  );
+
+export type SessionStatusReader = (
+  fiscalCode: FiscalCode
+) => TE.TaskEither<IResponseErrorInternal, UserSessionInfo>;
+
+export const getUserSessionStatusReader = (
+  sessionClient: SessionClient<"token">
+): SessionStatusReader => (fiscalCode): ReturnType<SessionStatusReader> =>
+  pipe(
+    TE.tryCatch(
+      async () => sessionClient.getSession({ fiscalcode: fiscalCode }),
```

Comment:
Do we need `async` here?

Reply:
I thought it could be useful for the abortable fetch, but I didn't try actually